paper.md

```diff
@@ -81,8 +81,12 @@ Let us take a relatively simple scenario where users have a list of single-purpo
 To accomplish this goal of running a list of containerized tasks using existing workflow engines, users need to learn a specific workflow language, deploy a workflow engine service, and learn to execute workflows on that service.
 These tasks may not be always trivial to accomplish if we assume the only thing users should care about is writing experimentation scripts and running them inside containers.
 Assume we have three scripts `download_dataset.py`, `verify_dataset.sh`, and `run_training.sh` to download a dataset, verify its contents and run a computational step.
-
-**TODO**: describe what we do **_without_** Popper, so that it becomes evident why we need to have Popper.
+In practice, when developers work following the container-native paradigm they end up interactively executing multiple Docker commands to build containers, compile code, test applications, deploy software, etc. 
+Keeping track of which commands were executed, in which order, and which flags were passed to each, can quickly become unmanageable, difficult to document, error prone, and hard to reproduce.
+The goal of Popper is to bring order to this chaotic scenario by providing a framework for clearly and explicitly defining container-native tasks.
+With Popper, users need to find or build Docker images containing the required dependencies, write a workflow in YAML, and execute a `popper run`.
+Popper will automate the process by running the scripts inside the specified containers as individual steps in the prescribed order.
+The workflow file can be shared through GitHub and the experiments can be reproduced in different computing environments with a single command.
```

Comment:
these three lines can be omitted as this is introduced later. The goal for this section is to talk about the problem that is being solved. The other thing we can talk about is that, if we want to run the same manual workflow but on another environment, the set of commands we execute end up being different. For example, singularity manages images differently (we can mention its cache, the flat image format, etc).

we can also expando on how running the same workflow but on a k8s or slurm means we will need to do distinct things